lib/tree.rb

```diff
@@ -1,38 +1,62 @@
 class NoApplesError < StandardError; end
 
-class AppleTree
-  attr_#fill_in :height, :age, :apples, :alive
+class Tree
+  attr_reader :height, :age, :apples, :alive
 
   def initialize
+    @height = 0
+    @age = 0
+    @apples = []
+    @alive = true
   end
 
   def age!
+    @age += 1
+    @height += 1
+    self.add_apples
+    if @age == 16
+      @alive = false
+    end
   end
 
   def add_apples
+    $i = 0
```

Comment:
You may have intended to use a local variable here, but that `$` prefix is making this a global instead. I think you can drop the `$`s in here.